muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -210,8 +262,20 @@ public Response toggleArchived(@PathParam ("NOTEID") Long noteId) {
       }
     }
 
-    Note updatedNote = notesController.toggleArchived(note);
+    note = notesController.toggleArchived(note);
     
-    return Response.ok(toRestModel(updatedNote)).build();
+    if(note.getArchived().equals(Boolean.FALSE)) {
```

Comment:
toggleArchived already sets these to null so this block shouldn't be needed here?

Also better to not change Entity's properties if there's no persist (i.e. outside a DAO).